core/src/main/java/com/datastrato/gravitino/connector/capability/Capability.java

```diff
@@ -93,6 +113,15 @@ public CapabilityResult caseSensitiveOnName(Scope scope) {
 
     @Override
     public CapabilityResult specificationOnName(Scope scope, String name) {
+      if (RESERVED_WORDS.contains(name.toLowerCase())) {
+        return CapabilityResult.unsupported(
+            String.format("The %s name '%s' is reserved.", scope, name));
+      }
+
+      if (!name.matches(LEGAL_PATTERN)) {
+        return CapabilityResult.unsupported(
+            String.format("The %s name '%s' is illegal.", scope, name));
+      }
```

Comment:
Can you please add more tests to test different names.

Reply:
added